### PR TITLE
js: fixes Argument type ChildNode is not assignable to parameter type Node

### PR DESCRIPTION
**Pull request by erikyo**
My contribution to a great cause, I'm just doing it for doing a little bit of my own part.

### CODE REVIEW

File: stand_with_ukraine.js
Comment (erikyo) [inline stand_with_ukraine.js]: const and let are supported after IE 11 - maybe is worth to support older browsers?
https://caniuse.com/const

**Comment by jonathanbossenger** [comment]
Thanks @erikyo 

@bueltge as the person who submitted #3, could I ask you to review this PR first, as a sanity check.

File: stand_with_ukraine.js
Comment (erikyo) [inline stand_with_ukraine.js]: Template string could be awesome but supported only [after IE](https://caniuse.com/?search=template%20string)
```div.innerHTML = `<a href="${swu_options.url}">${swu_options.text}</a>`;```

for the same reason as before I didn't know whether to use template string or leave as is.

**Comment by jonathanbossenger** [comment]
@rosswintle would you mind taking a peek at this PR as well, and leaving your thoughts on @erikyo's comments/suggestions.

**Review by bueltge — APPROVED** [review]

**Review by bueltge — APPROVED** [review]

File: stand_with_ukraine.js
Comment (rosswintle) [reply stand_with_ukraine.js]: Switching to `const` doesn't help support older browsers. `var` would be needed for that. Probably worth making that change.
```suggestion
	let div  = document.createElement('div');
```

File: stand_with_ukraine.js
Comment (rosswintle) [inline stand_with_ukraine.js]: `prepend` seems to not be supported in IE. So if we're aiming for support of older browsers this seems like it breaks support? Have I understood that right?

What does this change fix? Was something broken before?

I'm not seeing any errors.

**Comment by rosswintle** [comment]
Thanks for this contribution. Every suggestion is appreciated!

I'm not sure what this fixes though. Can you explain where you were seeing the error? What browser are you using? What theme is it showing in? If I can reproduce the error I can see if this change fixes something.

Personally, I prefer the original style, but it's @jonathanbossenger's plugin so I'll let him decide on the design. I like that you made the whole banner width clickable though.

There's a comment about IE11 compatibility but as far as I can tell the changes don't help with that. The code still uses `let`/`const`, and `prepend` has been added. I'd love for this to be IE11 compatibile but it looks like more changes are needed to make that happen.

Happy to merge if the design changes are OK. But I think more work is needed.

File: stand_with_ukraine.js
Comment (bueltge) [reply stand_with_ukraine.js]: https://developer.mozilla.org/en-US/docs/Web/API/Element/prepend
If we supported the old IE world, we should do not use `prepend` and leave the first idea.

**Comment by jonathanbossenger** [comment]
> Personally, I prefer the original style, but it's @jonathanbossenger's plugin so I'll let him decide on the design. I like that you made the whole banner width clickable though.
> 
Thanks @rosswintle I'll review the style later today, and leave my comments here.

File: stand_with_ukraine.js
Comment (erikyo) [reply stand_with_ukraine.js]: Another option could be:
`document.body.insertAdjacentElement('beforebegin', div);`
that works with IE > 5

**Comment by erikyo** [comment]
> I'm not sure what this fixes though. Can you explain where you were seeing the error? What browser are you using? What theme is it showing in? If I can reproduce the error I can see if this change fixes something.

The reason is that document.body is a Node and firstchild use in its constructor ChildNode (type ChildNode is not assignable to type Node). Detected using typescript console, because it does not really generate an error (ChildNode extends Node) but i guess isn't 100% correct
https://microsoft.github.io/PowerBI-JavaScript/interfaces/_node_modules_typedoc_node_modules_typescript_lib_lib_dom_d_.childnode.html#firstchild
https://dom.spec.whatwg.org/#childnode

About the style I noticed that it could be reduced and one property did not work ([text-combine](https://developer.mozilla.org/en-US/docs/Web/CSS/text-combine): #0057B8;) that's the main reason for the style changes. And ok, then I wanted to put my own spin on it, and made the banner look like the flag (and I was thought it looked awesome, but I understand it might be my own thing, degustibus non disputandum est - what you like isn't questionable :P)

image
image

File: stand_with_ukraine.js
Comment (erikyo) [reply stand_with_ukraine.js]: yes exactly, To be compatible with IE the first step was exactly that! 😅

File: stand-with-ukraine.php
Comment (erikyo) [inline stand-with-ukraine.php]: Again with the aim of squeezing the style, this could be removed completely as the text-decoration style isn't modified under normal circumstances

File: stand_with_ukraine.js
Comment (erikyo) [reply stand_with_ukraine.js]: you can try this test (also in the browser console of this page) which is much faster "insertAdjacentElement" just because the method argument is targeting the "right" element
```
const swu_options = {url:'',text:''}
function insertAE() {
  const div  = document.createElement('div');
  div.id = 'stand_with_ukraine_overlay';
  div.innerHTML = '<a href="' + swu_options.url + '">' + swu_options.text + '</a>';
  document.body.insertAdjacentElement('beforebegin', div);
}

function insertBE() {
  const body = document.getElementsByTagName('body')[0];
  const div  = document.createElement('div');
  div.id = 'stand_with_ukraine_overlay';
  div.innerHTML = '<a title="' + swu_options.text + '" href="' + swu_options.url + '">' + swu_options.hashtag + '</a>';
  document.body.insertBefore(div, body.firstChild);
}

performance.mark('mark-1')
Array.from(Array(5), insertBE)
Array.from(Array(5), insertAE)

performance.mark('mark-2')
Array.from(Array(1000), insertBE)

performance.mark('mark-3')
Array.from(Array(1000), insertAE)

performance.mark('mark-4')

// store test measurement
performance.measure('insertBefore', 'mark-2', 'mark-3' )
performance.measure('insertAdjacentElement2', 'mark-3', 'mark-4' )

// log results
console.log("insertBefore", performance.getEntriesByName('insertBefore')[0].duration)
console.log("insertAdjacentElement", performance.getEntriesByName('insertAdjacentElement2')[0].duration)
console.log(performance.getEntries())
```

**Comment by jonathanbossenger** [comment]
@erikyo I see there is a conflict with the `stand_with_ukraine.js` file. This is probably because I updated it after you created your PR. could I ask you to update your PR to resolve the conflict, then I can review it, thanks.

**Comment by erikyo** [comment]
no problem for that i can rebase the PR 😉

**Comment by erikyo** [comment]
then if you want I'll also pr for the style, but it was really wrong to include it in this one.

**Comment by jonathanbossenger** [comment]
> then if you want I'll also pr for the style, but it was really wrong to include it in this one.

That's no problem, you are welcome to fork the repo, and change the style to suit your preference.

**Comment by jonathanbossenger** [comment]
@erikyo for some reason, when I test the code from your branch, the font in the banner changes.

Code from the main branch: 
WordPress-Local-1

Code from your branch
WordPress-Local-2

Are you able to test this, and check why it might be the case?

**Comment by erikyo** [comment]
Yes you were right and the banners were not in the same place... and that's why they weren't look like the same. Fixed and now they should be perfectly identical. To test you can do this:
```
document.addEventListener('DOMContentLoaded', function() {
	let body = document.getElementsByTagName('body')[0];
	let div  = document.createElement('div');
	div.id = 'stand_with_ukraine_overlay';
	div.innerHTML = '<a title="' + swu_options.text + '" href="' + swu_options.url + '">' + swu_options.hashtag + '</a>';
	document.body.insertBefore(div, body.firstChild);
	

	var divNew  = document.createElement('div');
	divNew.id = 'stand_with_ukraine_overlay';
	divNew.innerHTML = '<a title="' + swu_options.text + '" href="' + swu_options.url + '">' + swu_options.hashtag + '</a>';
	document.body.insertAdjacentElement('afterbegin', divNew);
});
```

**Pull request closed without merging**
